docs/3.0/develop/transactions.mdx

```diff
@@ -141,45 +141,42 @@ from prefect.transactions import get_transaction, transaction
 
 
 @task
-def write_file(contents: str):
+def write_file(contents: str, filename: str):
     "Writes to a file."
-    with open("side-effect.txt", "w") as f:
+    with open(filename, "w") as f:
         f.write(contents)
 
 
 @write_file.on_rollback
-def del_file(transaction):
+def del_file(txn):
     "Deletes file."
-    os.unlink("side-effect.txt")
-    txn1 = get_transaction()      # get the transaction object
-    print(txn1.get("contents"))   # prints "hello world"
+    os.unlink(txn.get("file_name"))
 
 
 @task
-def quality_test():
+def quality_test(filename):
     "Checks contents of file."
-    with open("side-effect.txt", "r") as f:
+    with open(filename, "r") as f:
         data = f.readlines()
 
     if len(data) < 2:
-        txn2 = get_transaction()      # get the transaction object
-        raise ValueError(f"Not enough data! {(txn2.get("contents"))}") 
-        # prints "hello world" as part of the error message
+        raise ValueError(f"Not enough data!")
 
 
 @flow
 def pipeline(contents: str):
     with transaction() as txn:
-        write_file(contents)
-        sleep(2)
-        txn.set("contents", contents)  # set the contents key to the value of the contents variable
-        quality_test()
+        filename = "side-effect.txt"
+        txn.set("file_name", filename)
```

Comment:
i think these need to have the same time, so either both `file_name` or both `filename`